llvm/tools/llvm-dis/llvm-dis.cpp

```diff
@@ -259,6 +259,8 @@ int main(int argc, char **argv) {
       if (!DontPrint) {
         if (M) {
           ScopedDbgInfoFormatSetter FormatSetter(*M, WriteNewDbgInfoFormat);
+          if (WriteNewDbgInfoFormat)
+            M->removeDebugIntrinsicDeclarations();
```

Comment:
Should the conversion function unconditionally call `removeDebugIntrinsicDeclarations` instead, or does that cause problems with existing tests?

Reply:
It causes problems with existing tests, yes - when all the test IR gets updated (as we move to writing new-format by default) it'll be moved into the conversion function, but otherwise it causes intrinsic declarations to be reordered, which is awkward for tests and feels like a wrong behaviour.